src/gamescope/xwayland.rs

```diff
@@ -283,6 +283,15 @@ impl DBusInterface {
         Ok(value)
     }
 
+    /// Set the given window as an external overlay window
+    async fn get_external_overlay(&self, window_id: u32) -> fdo::Result<u32> {
+        self.ensure_connected().await;
+        self.xwayland
+            .get_external_overlay(window_id)
+            .map_err(|err| fdo::Error::Failed(err.to_string()))?;
+        Ok(window_id)
```

Comment:
You should return the value of the window property instead of the window id:

```suggestion
        let value = self.xwayland
            .get_external_overlay(window_id)
            .map_err(|err| fdo::Error::Failed(err.to_string()))?;
        Ok(value)
```